wordpress-mobile/WordPress-iOS
# Refresh the Plans feature

This PR refreshes the Plans feature and depends on a companion PR in WordPressKit.  Rather than a complete rewrite of Plans, for expediency I've opted to refactor only what was necessary to support the new endpoint and models. This leaves the original view model pattern mostly intact.  

The changes do the following: 
- Updates the plans and blog services to use the new endpoint.
- Refactors Plans to rely on core data for storage.
- Adds new models corresponding to the data returned by the new remote endpoint. 
- Removes icons from plan feature descriptions. 
- Feature descriptions are now a flat list rather than sectioned. 
- Removes old purchasing related code that was not being used.

After the changes plans appears as you see here:
#### Plans List
![simulator screen shot - iphone 8 - 2018-12-19 at 18 17 39](https://user-images.githubusercontent.com/1435271/50256718-3dad9880-03bd-11e9-88ac-d9b70bcdcbe5.png)
#### Plans Detail
![simulator screen shot - iphone 8 - 2018-12-19 at 18 17 54](https://user-images.githubusercontent.com/1435271/50256726-44d4a680-03bd-11e9-8738-bf7c95d2577f.png)

Note that the icons shown are determined by the API endpoint. Currently the icons for the blogger and ecommerce plans are the same as the free and business plans, but will be updated separate from this PR.

@iamthomasbishop, we chatted a while back about UI changes to the feature. We've ended up with only six plans to show rather than the eight we thought we'd have, and since one plan appears under personal and business groups I've opted to just use a flat list as that seems to make more sense than sections.  Apart from that change I'm hopeful this captures what we discussed but please let me know if I've overlooked something. 

To test:
- Build and test the app.
- Confirm that the Plans feature loads for wpcom sites.
- Confirm that there are six plans available (no dups ;) ).
- Confirm that icons are removed from the feature descriptions in the plan detail screen and that feature text is not truncated.
- Confirm layout does not break when rotating. 
- Confirm layout looks correct on the iPad.
- For paranoia, test the upgrade path and ensure there is no crash accessing plans after an upgrade.

Update release notes:


This PR is 1,184 additions but over 800 are due to core data changes so hopefully not too much to review 🤞.  (We don't count those lines right?  Right? 😬 ) 

@jklausa hoping you can look at this one as well.
